lib/ui/painting.dart

```diff
@@ -2944,6 +3009,28 @@ class _ImageFilter extends NativeFieldWrapperClass2 {
   }
   void _initMatrix(Float64List matrix4, int filterQuality) native 'ImageFilter_initMatrix';
 
+  /// Converts a color filter to an image filter.
+  _ImageFilter.fromColorFilter(this.creator)
+    : assert(creator != null), // ignore: unnecessary_null_comparison
+      assert(creator._type == ImageFilter._kTypeColorFilter) {
+    _constructor();
+    final _ColorFilter? nativeFilter = creator._colorFilter?._toNativeColorFilter();
+    assert(nativeFilter != null);
+    _initColorFilter(nativeFilter!);
+  }
+  void _initColorFilter(_ColorFilter colorFilter) native 'ImageFilter_initColorFilter';
+
+  /// Composes `_innerFilter` with `_outerFilter`.
+  _ImageFilter.composed(this.creator)
+    : assert(creator != null), // ignore: unnecessary_null_comparison
+      assert(creator._type == ImageFilter._kTypeComposed) {
+    _constructor();
+    final _ImageFilter? nativeFilterInner = creator._innerFilter?._toNativeImageFilter();
+    final _ImageFilter? nativeFilterOuter = creator._outerFilter?._toNativeImageFilter();
+    assert(nativeFilterInner != null && nativeFilterOuter != null);
+    _initComposed(nativeFilterOuter!,  nativeFilterInner!);
```

Comment:
The skia function allows either filter to be `nullptr` it seems. But I'm not sure how to achieve this in dart now that nnbd is enabled (My test app segfaulted when I changed the signature of `_initComposed` to `void Function(_ImageFilter?, _ImageFilter?)`) .

Reply:
The Constructor here doesn't allow for nulls, so is this just theoretical?

Also, the native methods don't look like they check the passed down parameters for null before they invoke a getter on them - could that be causing the segfault?

Reply:
Ahh that makes sense I must be dereferencing a null pointer.

Reply:
Turned out it was indeed the cause. The API now makes more sense (no longer throws when a null color filter is used). Thank you!